include/aws/http/request_response.h

```diff
@@ -546,6 +600,28 @@ struct aws_input_stream *aws_http_message_get_body_stream(const struct aws_http_
 AWS_HTTP_API
 void aws_http_message_set_body_stream(struct aws_http_message *message, struct aws_input_stream *body_stream);
 
+/**
+ * Submit a chunk of data to a stream for writing to an outbound HTTP/1.1 connection using chunked transfer encoding.
+ * Note: The stream does not take ownership of allocated resources in the chunk, if any. The caller is responsible
+ * for managing these resources. The data inside the chunk is expected not to be modified by the caller until
+ * the on_complete callback inside the chunk structure is invoked (if provided by the caller). Upon invocation of
+ * on_complete, the caller may release or modify structures in the chunk.
+ * On successful submission of the chunk to the stream, AWS_OP_SUCCESS is returned and AWS_OP_ERR otherwise.
+ */
+AWS_HTTP_API int aws_http1_stream_write_chunk(struct aws_http_stream *stream, struct aws_http1_stream_chunk *chunk);
```

Comment:
The usual approach we've taken is for the public function to take the `options`, and then we allocate internal book-keeping stuff (ex: `struct aws_http1_stream_chunk`) based on that. It's a little weird for us to ask the user to give us the stuff we'll need for our internal book-keeping.

It makes it challenging to change the internal implementation when too much is public

Reply:
I now manage the internal book keeping entirely in h1_stream.c and h1_encoder.c. The user is now no longer aware of the chunk.